### PR TITLE
Replaced checkbox demos with reactive form examples

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-forms?_sp.spa=pr-checkbox-demos](https://developer.blackbaud.com/skyux-forms?_sp.spa=pr-checkbox-demos)

File: src/app/public/plugin-resources/code-examples/checkbox/basic/checkbox-demo.component.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<form
```
Comment (blackbaud-johnly) [inline src/app/public/plugin-resources/code-examples/checkbox/basic/checkbox-demo.component.html:1]: When I open this in Stackblitz and select the checkboxes, the color changes to indicate that they are selected but they don't have the checkmarks inside the checkboxes. Is that the expected behavior?

File: src/app/docs/checkbox/checkbox-docs.component.html [hunk 2]
```diff
@@ -61,7 +61,13 @@ <h4>
 
   <sky-docs-code-examples>
     <sky-docs-code-example
-      sourceCodePath="src/app/public/plugin-resources/code-examples/checkbox"
+      heading="Basic"
```
Comment (blackbaud-johnly) [inline src/app/docs/checkbox/checkbox-docs.component.html:64]:
```suggestion
      heading="Standard checkboxes"
```

File: src/app/docs/checkbox/checkbox-docs.component.html [hunk 3]
```diff
@@ -61,7 +61,13 @@ <h4>
...
+      sourceCodePath="src/app/public/plugin-resources/code-examples/checkbox/basic"
+    >
+    </sky-docs-code-example>
+    <sky-docs-code-example
+      heading="Icon group"
```
Comment (blackbaud-johnly) [inline src/app/docs/checkbox/checkbox-docs.component.html:69]:
```suggestion
      heading="Icon checkbox group"
```

File: src/app/public/plugin-resources/code-examples/checkbox/icon-group/checkbox-demo.component.html [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<form
+  [formGroup]="myForm"
+  (ngSubmit)="onSubmit()"
+>
+  <div
+    class="sky-switch-icon-group sky-margin-stacked-default"
+  >
+    <sky-checkbox
+      formControlName="bold"
+      icon="bold"
```
[the remark this answers is not in the document]
Comment (Blackbaud-AlexKingman) [reply src/app/public/plugin-resources/code-examples/checkbox/icon-group/checkbox-demo.component.html:10]: Yeah its a known issue with checkbox. We need to look into in a separate bucket of work.

File: src/app/public/plugin-resources/code-examples/checkbox/basic/checkbox-demo.component.html (hunk 1 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/plugin-resources/code-examples/checkbox/basic/checkbox-demo.component.html:1]: Known issue - see answer below.

File: src/app/public/plugin-resources/code-examples/checkbox/icon-group/checkbox-demo.component.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (blackbaud-johnly) [reply src/app/public/plugin-resources/code-examples/checkbox/icon-group/checkbox-demo.component.html:10]: Sounds good. Didn't realize there was a known issue. Thanks!

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-forms?_sp.spa=pr-checkbox-demos](https://developer.blackbaud.com/skyux-forms?_sp.spa=pr-checkbox-demos)

**Review by blackbaud-johnly — APPROVED** [review]

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**